Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -384,6 +395,27 @@ protected override async Task OnAfterRenderAsync( bool firstRender )
         await base.OnAfterRenderAsync( firstRender );
     }
 
+    private void AutoGenerateColumns()
+    {
+        var properties = ReflectionHelper.GetPublicProperties<TItem>();
+        foreach ( var property in properties )
+        {
+            if ( !( property.PropertyType.IsValueType || property.PropertyType == typeof( string ) ) )
+            {
+                continue;
+            }
+
+            var column = new DataGridColumn<TItem>
+            {
+                Caption = ReflectionHelper.ResolveCaption( property ),
+                Field = property.Name,

```

Comment:
What else might we get from the property?

-` [Required]` attribute?
- Enum that generates the Select with options?

Reply:
>  [Required] attribute? 
I'm blanking out here, what can we set in the Column with the Required attribute?

> Enum that generates the Select with options?
What's your idea with Enum? If an Enum is found we figure out the Enum members and generate the Select, that's it right? No need for Attributes here, right?

Reply:
> I'm blanking out here, what can we set in the Column with the Required attribute?
> 
> >

Basically, validation.

Reply:
> > I'm blanking out here, what can we set in the Column with the Required attribute?
> > >
> 
> Basically, validation.

I still don't understand what you want to do here, there's no validation atributes that we set specifically on the column is there? The `DataAnnotations` validation should work automatically if the `DataGrid` is set to validate, we've talked about this.

Reply:
You're right. I completely forgot that we mentioned it last time.